tests/DotNetLightning.Core.Tests/TransactionTests.fs

```diff
@@ -326,4 +329,551 @@ let testList = testList "transaction tests" [
             
         ()
     *)
+    testCase "check spend handling from local/remote commitment txs" <| fun _ ->
+        let rand = Random()
+
+        let localNodeMasterPrivKey =
+            let extKey = ExtKey()
+            NodeMasterPrivKey extKey
+        let localChannelPrivKeys = localNodeMasterPrivKey.ChannelPrivKeys (rand.Next(1, 100))
+        let localChannelPubKeys = localChannelPrivKeys.ToChannelPubKeys()
+
+        let remoteNodeMasterPrivKey =
+            let extKey = ExtKey()
+            NodeMasterPrivKey extKey
+        let remoteChannelPrivKeys = remoteNodeMasterPrivKey.ChannelPrivKeys (rand.Next(1, 100))
+        let remoteChannelPubKeys = remoteChannelPrivKeys.ToChannelPubKeys()
+
+        let fundingAmount = 10_000_000L |> Money.Satoshis
+        let fundingScriptPubKey =
+            Scripts.funding
+                localChannelPubKeys.FundingPubKey
+                remoteChannelPubKeys.FundingPubKey
+        let fundingDestination = fundingScriptPubKey.WitHash :> IDestination
```

Comment:
It is not super-important, but I want you to have a small scope for temporary variable like this.
i.e., instead of writing
```fsharp
let x = 1
let y = 2
let z = x + y
```

write

```fsharp
let z =
    let x = 1
    let y = 2
    x + y
```

In this way I can understand the lifetime of `x`, `y` more easily and it gets easy to review. (This is one of the biggest reason I use F#)

In this case, `fundingScriptCode` is what we want to have later. So

```fsharp

let fundingScriptCoin =
    let fundingDestination = // ..
    // ...
    ScriptCoin(fundingCoin, fundingScriptPubKey)
```

Reply:
love that! I wonder if we could write an FSharpLint rule that could suggest this

Reply:
Done.